ecosystem/node-checker/src/evaluators/direct/api/transaction_presence.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{super::DirectEvaluatorInput, ApiEvaluatorError, API_CATEGORY};
+use crate::{
+    configuration::EvaluatorArgs,
+    evaluator::{EvaluationResult, Evaluator},
+    evaluators::EvaluatorType,
+};
+use anyhow::{anyhow, Result};
+use aptos_rest_client::{aptos_api_types::TransactionInfo, Client as AptosRestClient, Transaction};
+use aptos_sdk::crypto::HashValue;
+use clap::Parser;
+use poem_openapi::Object as PoemObject;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+#[derive(Clone, Debug, Deserialize, Parser, PoemObject, Serialize)]
+pub struct TransactionPresenceEvaluatorArgs {
+    #[clap(long, default_value_t = 5)]
+    pub transaction_fetch_delay_secs: u64,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct TransactionPresenceEvaluator {
+    args: TransactionPresenceEvaluatorArgs,
+}
+
+impl TransactionPresenceEvaluator {
+    pub fn new(args: TransactionPresenceEvaluatorArgs) -> Self {
+        Self { args }
+    }
+
+    /// Get the transaction info of the latest transaction.
+    async fn get_transaction_info(
+        client: &AptosRestClient,
+    ) -> Result<TransactionInfo, ApiEvaluatorError> {
+        Self::unwrap_transaction_info(
+            client
+                .get_transactions(None, Some(1))
```

Comment:
Never checked this, but if you don't specify it, what does it default to? 

Given that this evaluator is focussed on data availability and not synced versions, I feel like a better approach might be to get the latest transaction at the node under test (i.e., at the highest version), so we don't need to worry about the pruner and then compare that to the transaction returned by the baseline node. Also, I think we're already assuming that the baseline node is always more up-to-date, so we can fetch the data from both nodes and compare them safely?

Reply:
My concern with this direction is the target node could return a bogus transaction (either bad fields or it just doesn't exist) and then it'll be hard for us to know whether that's the issue or something is wrong with our baseline.

Reply:
I'm not following you... 🤔 My concern with this check is that I don't know which version will be fetched by
`client.get_transactions(None, Some(1))` because you haven't specified a version. What does it default to? 0, or the latest? Both are problematic due to the pruner, state sync mode, sync latency between baseline and target. 

> My concern with this direction is the target node could return a bogus transaction (either bad fields or it just doesn't exist) and then it'll be hard for us to know whether that's the issue or something is wrong with our baseline.

We'll still be doing the same checks as above, just with the version set at the highest of the target node. Will ping you offline to understand the miscommunication.